crates/aptos-telemetry-service/src/prometheus_push_metrics.rs

```diff
@@ -56,13 +78,20 @@ pub async fn handle_metrics_ingest(
     };
 
     let start_timer = Instant::now();
+    let successful_clients = aptos_infallible::RwLock::new(Vec::<String>::new());
```

Comment:
We should use a futures lock here which supports async/await. With this kind of lock, we would end up sleeping the current thread, which means we might be out of threads to run other concurrent futures. With async/await the tokio runtime will schedule another future on the thread instead. Important in this case because we are joining on futures here.